linux_test.go

```diff
@@ -292,3 +292,68 @@ func TestFileModeIsSet(t *testing.T) {
 		equals(c.isSet, l.fileModeIsSet(), t)
 	}
 }
+
+func testChangeModeOnRotate(t *testing.T, oldMode fs.FileMode, newMode fs.FileMode) {
+	currentTime = fakeTime
+	dir := t.TempDir()
+	defer os.RemoveAll(dir)
+
+	// create logger
+	filename := logFile(dir)
+	l := &Logger{
+		Filename:   filename,
+		MaxBackups: 1,
+		MaxSize:    100, // megabytes
+		FileMode:   oldMode,
+	}
+	defer l.Close()
+
+	// create file
+	mode := defaultFileMode
+	if l.fileModeIsSet() {
+		mode = l.FileMode
+	}
+	f, err := os.OpenFile(filename, os.O_CREATE|os.O_RDWR, mode)
+	isNil(err, t)
+	f.Close()
+
+	// change logger file mode
+	l.FileMode = newMode
```

Comment:
Why do you set this? I would have expected a simple test, you write a file with default mode, no need to log to it, then you create the new Logger with corresponding FileMode: newMode, you trigger a write which should trigger a rotate, then at least: `equals(newMode, info.Mode(), t)`  ? am I missing something obvious?

On another note what happens to the main tetragon.log file, do we explicitly chown that file with new mode or not? I can't recall. If not while you are it please ensure if the main non-rotated file exist then we do a proper chown on it using the permission passed from tetragon so we can do 0644<=>0600

Reply:
I added tests for four transitions (L355-358): default->640, 640->default (this one is a no-op), 600->644 and 644->600. I wanted to check that the permissions change doesn't happen to early (e.g. on write), only on rotate, that's why I first change the logger permissions then write to the file. Does it make sense?

> On another note what happens to the main tetragon.log file, do we explicitly chown that file with new mode or not? I can't recall.

Yes, we explicitly chown in `openNew()`. So if the non-rotated file exists then we do a proper chown on it using given permissions when it gets rotated.